### PR TITLE
Update Orca Data Documentation

### PR DESCRIPTION
**Pull request by jenniew**
Add python docstring for Orca data APIs.
Add more API documents to Orca data programming guide.

### CODE REVIEW

File: docs/docs/ProgrammingGuide/orca-data.md [hunk 1]
```diff
@@ -1,44 +1,36 @@
 ---
 ## **Introduction**
 
-Analytics Zoo orca data provides data-parallel pre-processing support for AI.
+Analytics Zoo Orca data provides data-parallel pre-processing support for Python AI.
 
-It supports data pre-processing from different data sources, like TensorFlow DataSet, PyTorch DataLoader, MXNet DataLoader, etc. and it supports different data format, like Pandas DataFrame, Numpy, Images, Parquet.
+It supports data pre-processing from different data sources, like TensorFlow DataSet, PyTorch DataLoader, MXNet DataLoader, etc. and it supports various data format, like Pandas DataFrame, Numpy, Images, Parquet.
```
Comment (hkvision) [inline docs/docs/ProgrammingGuide/orca-data.md:6]: format -> formats
add ,etc at the end.

File: docs/docs/ProgrammingGuide/orca-data.md [hunk 2]
```diff
@@ -1,44 +1,36 @@
...
-The backend distributed preprocessing engine can be [Spark](https://spark.apache.org/) or [Ray](https://github.com/ray-project/ray).
-
-In current version, orca data API only supports parallel pre-processing with Pandas DataFrame on Ray.
+The distributed backend engine can be [Spark](https://spark.apache.org/) or [Ray](https://github.com/ray-project/ray). We now support Spark-based transformations to do the pre-processing, and provide functionality to seamlessly put data to Ray cluster for later training/inference on Ray. 
 
 ---
 ## **XShards**
 
-XShards is a collection of data in orca data API. In current version, the element in XShards is a Pandas DataFrame.
+XShards is a collection of data in Orca data API. For different backend, we have two XShards: SparkXShards and RayXShards.
```
Comment (hkvision) [inline docs/docs/ProgrammingGuide/orca-data.md:13]: I don't think we need to say RayXShards here... Not supposed to read by users...

File: pyzoo/zoo/orca/data/shard.py [hunk 3]
```diff
@@ -202,19 +205,37 @@ def __init__(self, rdd, transient=False):
         self.type = {}
 
     def transform_shard(self, func, *args):
+        """
+        Return a new SparkXShards by applying a function to each shard of this SparkXShards
+        :param func: python function to process data. The first argument is the data shard.
+        :param args: other arguments in this function.
+        :return:
```
Comment (hkvision) [inline pyzoo/zoo/orca/data/shard.py:212]: ??? Either add something or remove this

File: docs/docs/ProgrammingGuide/orca-data.md (hunk 2 above)
Comment (jenniew) [reply docs/docs/ProgrammingGuide/orca-data.md]: updated.

**Comment by jenniew** [comment]
Jenkins pass: http://10.239.47.210:18888/job/ZOO-PR-Validation/4086/

**Pull request merged**